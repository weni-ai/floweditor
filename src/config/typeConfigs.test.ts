import {
  getTypeConfig,
  typeConfigList,
  typeConfigMap,
} from 'config/typeConfigs';

describe('typeConfigs', () => {
  it('should provide type configs', () => {
    expect(typeConfigList).toMatchSnapshot();
    expect(typeConfigMap).toMatchSnapshot();

    Object.keys(typeConfigMap).forEach((key: string) =>
      expect(getTypeConfig(key as any)).toEqual(typeConfigMap[key]),
    );
  });
});
