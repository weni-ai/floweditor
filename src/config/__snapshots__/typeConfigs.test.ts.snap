// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typeConfigs should provide type configs 1`] = `
Array [
  Object {
    "component": [Function],
    "description": " ** Unsupported ** ",
    "name": "Missing",
    "type": "missing",
    "visibility": Array [
      "-",
    ],
  },
  Object {
    "component": [Function],
    "description": "Play a message",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "text",
      "audio_url",
    ],
    "name": "Play Message",
    "type": "say_msg",
    "visibility": Array [
      "voice",
    ],
  },
  Object {
    "description": "Wait for menu selection",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Wait for Menu Selection",
    "type": "wait_for_menu",
    "visibility": Array [
      "voice",
    ],
  },
  Object {
    "description": "Wait for multiple digits",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
      "cases",
    ],
    "name": "Wait for Digits",
    "type": "wait_for_digits",
    "visibility": Array [
      "voice",
    ],
  },
  Object {
    "description": "Wait for an audio recording",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Wait for Audio",
    "type": "wait_for_audio",
    "visibility": Array [
      "messaging_offline",
      "voice",
    ],
  },
  Object {
    "description": "Wait for forwarded call to someone else",
    "form": [Function],
    "name": "Wait for Forwarded Call",
    "type": "wait_for_dial",
    "visibility": Array [
      "voice",
    ],
  },
  Object {
    "component": [Function],
    "description": "Send the contact a message",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "text",
      "quick_replies",
      "templating.variables",
    ],
    "massageForDisplay": [Function],
    "name": "Send message",
    "type": "send_msg",
  },
  Object {
    "aliases": Array [
      "split_by_whatsapp_product",
    ],
    "component": [Function],
    "description": "Send a WhatsApp product",
    "filter": "whatsapp_catalog",
    "form": [Function],
    "name": "Send WhatsApp Product",
    "new": true,
    "type": "send_msg_catalog",
  },
  Object {
    "aliases": Array [
      "switch",
    ],
    "description": "Wait for the contact to respond",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "categories",
      "cases",
    ],
    "name": "Wait for response",
    "type": "wait_for_response",
    "visibility": Array [
      "messaging",
      "messaging_offline",
    ],
  },
  Object {
    "aliases": Array [
      "smart",
    ],
    "description": "Smartly wait for response",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "categories",
      "cases",
    ],
    "name": "Smart Wait for Response",
    "new": true,
    "type": "smart_wait_for_response",
    "visibility": Array [
      "messaging",
      "messaging_offline",
    ],
  },
  Object {
    "aliases": Array [
      "split_by_wenigpt",
    ],
    "component": [Function],
    "description": "Call WeniGPT",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "input",
    ],
    "name": "Call WeniGPT",
    "new": true,
    "type": "call_wenigpt",
  },
  Object {
    "component": [Function],
    "description": "Send somebody else a message",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "text",
    ],
    "name": "Send broadcast",
    "type": "send_broadcast",
  },
  Object {
    "component": [Function],
    "description": "Label the incoming message",
    "form": [Function],
    "name": "Add label",
    "type": "add_input_labels",
    "visibility": Array [
      "messaging",
      "messaging_offline",
      "voice",
    ],
  },
  Object {
    "component": [Function],
    "description": "Add a URN for the contact",
    "form": [Function],
    "name": "Add URN",
    "type": "add_contact_urn",
  },
  Object {
    "component": [Function],
    "description": "Add the contact to a group",
    "form": [Function],
    "name": "Add to group",
    "type": "add_contact_groups",
  },
  Object {
    "component": [Function],
    "description": "Remove the contact from a group",
    "form": [Function],
    "name": "Remove from group",
    "type": "remove_contact_groups",
  },
  Object {
    "aliases": Array [
      "set_contact_name",
      "set_contact_language",
      "set_contact_channel",
      "set_contact_status",
    ],
    "component": [Function],
    "description": "Update the contact",
    "form": [Function],
    "name": "Update contact",
    "type": "set_contact_field",
  },
  Object {
    "component": [Function],
    "description": "Send email",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "subject",
      "body",
    ],
    "name": "Send email",
    "type": "send_email",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  Object {
    "component": [Function],
    "description": "Save a result for this flow",
    "form": [Function],
    "name": "Save flow result",
    "type": "set_run_result",
  },
  Object {
    "component": [Function],
    "description": "Play a contact recording",
    "form": [Function],
    "name": "Play Recording",
    "type": "play_audio",
    "visibility": Array [
      "voice",
    ],
  },
  Object {
    "aliases": Array [
      "split_by_webhook",
    ],
    "component": [Function],
    "description": "Call a webhook",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Call webhook",
    "type": "call_webhook",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  Object {
    "aliases": Array [
      "split_by_resthook",
    ],
    "component": [Function],
    "description": "Call Zapier",
    "filter": "resthook",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Call Zapier",
    "type": "call_resthook",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  Object {
    "aliases": Array [
      "split_by_subflow",
    ],
    "component": [Function],
    "description": "Enter another flow",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Enter another flow",
    "type": "enter_flow",
  },
  Object {
    "component": [Function],
    "description": "Start somebody else in a flow",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Start somebody else",
    "type": "start_session",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  Object {
    "aliases": Array [
      "split_by_ticket",
    ],
    "component": [Function],
    "description": "Open a ticket with a human agent",
    "filter": "ticketer",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Open Ticket",
    "type": "open_ticket",
  },
  Object {
    "aliases": Array [
      "split_by_airtime",
    ],
    "component": [Function],
    "description": "Send the contact airtime",
    "filter": "airtime",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Send Airtime",
    "type": "transfer_airtime",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  Object {
    "aliases": Array [
      "split_by_external_service",
    ],
    "component": [Function],
    "description": "Call an external service",
    "filter": "external_service",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exists",
    ],
    "name": "Call External Service",
    "type": "call_external_service",
  },
  Object {
    "description": "Wait for an image",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Wait for Image",
    "type": "wait_for_image",
    "visibility": Array [
      "messaging_offline",
    ],
  },
  Object {
    "description": "Wait for a video",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Wait for Video",
    "type": "wait_for_video",
    "visibility": Array [
      "messaging_offline",
    ],
  },
  Object {
    "description": "Wait for location GPS coordinates",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Wait for Location",
    "type": "wait_for_location",
    "visibility": Array [
      "messaging_offline",
    ],
  },
  Object {
    "aliases": Array [
      "call_classifier",
    ],
    "component": [Function],
    "description": "Split by intent",
    "filter": "classifier",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Split by intent",
    "type": "split_by_intent",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  Object {
    "aliases": Array [
      "smart",
    ],
    "description": "Automatic Classify",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "categories",
      "cases",
    ],
    "name": "Automatic Classify",
    "new": true,
    "type": "automatic_classify",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  Object {
    "description": "Split by a custom expression",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
      "cases",
    ],
    "name": "Split by expression",
    "type": "split_by_expression",
  },
  Object {
    "description": "Split by contact field",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
      "cases",
    ],
    "name": "Split by contact field",
    "type": "split_by_contact_field",
  },
  Object {
    "aliases": Array [
      "split_by_run_result_delimited",
    ],
    "description": "Split by a result in the flow",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
      "cases",
    ],
    "name": "Split by flow result",
    "type": "split_by_run_result",
  },
  Object {
    "description": "Split randomly",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Split randomly",
    "type": "split_by_random",
  },
  Object {
    "description": "Split by group membership",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Split by group membership",
    "type": "split_by_groups",
  },
  Object {
    "description": "Split by URN type",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Split by URN type",
    "type": "split_by_scheme",
  },
]
`;

exports[`typeConfigs should provide type configs 2`] = `
Object {
  "add_contact_groups": Object {
    "component": [Function],
    "description": "Add the contact to a group",
    "form": [Function],
    "name": "Add to group",
    "type": "add_contact_groups",
  },
  "add_contact_urn": Object {
    "component": [Function],
    "description": "Add a URN for the contact",
    "form": [Function],
    "name": "Add URN",
    "type": "add_contact_urn",
  },
  "add_input_labels": Object {
    "component": [Function],
    "description": "Label the incoming message",
    "form": [Function],
    "name": "Add label",
    "type": "add_input_labels",
    "visibility": Array [
      "messaging",
      "messaging_offline",
      "voice",
    ],
  },
  "automatic_classify": Object {
    "aliases": Array [
      "smart",
    ],
    "description": "Automatic Classify",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "categories",
      "cases",
    ],
    "name": "Automatic Classify",
    "new": true,
    "type": "automatic_classify",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  "call_classifier": Object {
    "aliases": Array [
      "call_classifier",
    ],
    "component": [Function],
    "description": "Split by intent",
    "filter": "classifier",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Split by intent",
    "type": "split_by_intent",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  "call_external_service": Object {
    "aliases": Array [
      "split_by_external_service",
    ],
    "component": [Function],
    "description": "Call an external service",
    "filter": "external_service",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exists",
    ],
    "name": "Call External Service",
    "type": "call_external_service",
  },
  "call_resthook": Object {
    "aliases": Array [
      "split_by_resthook",
    ],
    "component": [Function],
    "description": "Call Zapier",
    "filter": "resthook",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Call Zapier",
    "type": "call_resthook",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  "call_webhook": Object {
    "aliases": Array [
      "split_by_webhook",
    ],
    "component": [Function],
    "description": "Call a webhook",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Call webhook",
    "type": "call_webhook",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  "call_wenigpt": Object {
    "aliases": Array [
      "split_by_wenigpt",
    ],
    "component": [Function],
    "description": "Call WeniGPT",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "input",
    ],
    "name": "Call WeniGPT",
    "new": true,
    "type": "call_wenigpt",
  },
  "enter_flow": Object {
    "aliases": Array [
      "split_by_subflow",
    ],
    "component": [Function],
    "description": "Enter another flow",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Enter another flow",
    "type": "enter_flow",
  },
  "missing": Object {
    "component": [Function],
    "description": " ** Unsupported ** ",
    "name": "Missing",
    "type": "missing",
    "visibility": Array [
      "-",
    ],
  },
  "open_ticket": Object {
    "aliases": Array [
      "split_by_ticket",
    ],
    "component": [Function],
    "description": "Open a ticket with a human agent",
    "filter": "ticketer",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Open Ticket",
    "type": "open_ticket",
  },
  "play_audio": Object {
    "component": [Function],
    "description": "Play a contact recording",
    "form": [Function],
    "name": "Play Recording",
    "type": "play_audio",
    "visibility": Array [
      "voice",
    ],
  },
  "remove_contact_groups": Object {
    "component": [Function],
    "description": "Remove the contact from a group",
    "form": [Function],
    "name": "Remove from group",
    "type": "remove_contact_groups",
  },
  "say_msg": Object {
    "component": [Function],
    "description": "Play a message",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "text",
      "audio_url",
    ],
    "name": "Play Message",
    "type": "say_msg",
    "visibility": Array [
      "voice",
    ],
  },
  "send_broadcast": Object {
    "component": [Function],
    "description": "Send somebody else a message",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "text",
    ],
    "name": "Send broadcast",
    "type": "send_broadcast",
  },
  "send_email": Object {
    "component": [Function],
    "description": "Send email",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "subject",
      "body",
    ],
    "name": "Send email",
    "type": "send_email",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  "send_msg": Object {
    "component": [Function],
    "description": "Send the contact a message",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "text",
      "quick_replies",
      "templating.variables",
    ],
    "massageForDisplay": [Function],
    "name": "Send message",
    "type": "send_msg",
  },
  "send_msg_catalog": Object {
    "aliases": Array [
      "split_by_whatsapp_product",
    ],
    "component": [Function],
    "description": "Send a WhatsApp product",
    "filter": "whatsapp_catalog",
    "form": [Function],
    "name": "Send WhatsApp Product",
    "new": true,
    "type": "send_msg_catalog",
  },
  "set_contact_channel": Object {
    "aliases": Array [
      "set_contact_name",
      "set_contact_language",
      "set_contact_channel",
      "set_contact_status",
    ],
    "component": [Function],
    "description": "Update the contact",
    "form": [Function],
    "name": "Update contact",
    "type": "set_contact_field",
  },
  "set_contact_field": Object {
    "aliases": Array [
      "set_contact_name",
      "set_contact_language",
      "set_contact_channel",
      "set_contact_status",
    ],
    "component": [Function],
    "description": "Update the contact",
    "form": [Function],
    "name": "Update contact",
    "type": "set_contact_field",
  },
  "set_contact_language": Object {
    "aliases": Array [
      "set_contact_name",
      "set_contact_language",
      "set_contact_channel",
      "set_contact_status",
    ],
    "component": [Function],
    "description": "Update the contact",
    "form": [Function],
    "name": "Update contact",
    "type": "set_contact_field",
  },
  "set_contact_name": Object {
    "aliases": Array [
      "set_contact_name",
      "set_contact_language",
      "set_contact_channel",
      "set_contact_status",
    ],
    "component": [Function],
    "description": "Update the contact",
    "form": [Function],
    "name": "Update contact",
    "type": "set_contact_field",
  },
  "set_contact_status": Object {
    "aliases": Array [
      "set_contact_name",
      "set_contact_language",
      "set_contact_channel",
      "set_contact_status",
    ],
    "component": [Function],
    "description": "Update the contact",
    "form": [Function],
    "name": "Update contact",
    "type": "set_contact_field",
  },
  "set_run_result": Object {
    "component": [Function],
    "description": "Save a result for this flow",
    "form": [Function],
    "name": "Save flow result",
    "type": "set_run_result",
  },
  "smart": Object {
    "aliases": Array [
      "smart",
    ],
    "description": "Automatic Classify",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "categories",
      "cases",
    ],
    "name": "Automatic Classify",
    "new": true,
    "type": "automatic_classify",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  "smart_wait_for_response": Object {
    "aliases": Array [
      "smart",
    ],
    "description": "Smartly wait for response",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "categories",
      "cases",
    ],
    "name": "Smart Wait for Response",
    "new": true,
    "type": "smart_wait_for_response",
    "visibility": Array [
      "messaging",
      "messaging_offline",
    ],
  },
  "split_by_airtime": Object {
    "aliases": Array [
      "split_by_airtime",
    ],
    "component": [Function],
    "description": "Send the contact airtime",
    "filter": "airtime",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Send Airtime",
    "type": "transfer_airtime",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  "split_by_contact_field": Object {
    "description": "Split by contact field",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
      "cases",
    ],
    "name": "Split by contact field",
    "type": "split_by_contact_field",
  },
  "split_by_expression": Object {
    "description": "Split by a custom expression",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
      "cases",
    ],
    "name": "Split by expression",
    "type": "split_by_expression",
  },
  "split_by_external_service": Object {
    "aliases": Array [
      "split_by_external_service",
    ],
    "component": [Function],
    "description": "Call an external service",
    "filter": "external_service",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exists",
    ],
    "name": "Call External Service",
    "type": "call_external_service",
  },
  "split_by_groups": Object {
    "description": "Split by group membership",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Split by group membership",
    "type": "split_by_groups",
  },
  "split_by_intent": Object {
    "aliases": Array [
      "call_classifier",
    ],
    "component": [Function],
    "description": "Split by intent",
    "filter": "classifier",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Split by intent",
    "type": "split_by_intent",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  "split_by_random": Object {
    "description": "Split randomly",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Split randomly",
    "type": "split_by_random",
  },
  "split_by_resthook": Object {
    "aliases": Array [
      "split_by_resthook",
    ],
    "component": [Function],
    "description": "Call Zapier",
    "filter": "resthook",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Call Zapier",
    "type": "call_resthook",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  "split_by_run_result": Object {
    "aliases": Array [
      "split_by_run_result_delimited",
    ],
    "description": "Split by a result in the flow",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
      "cases",
    ],
    "name": "Split by flow result",
    "type": "split_by_run_result",
  },
  "split_by_run_result_delimited": Object {
    "aliases": Array [
      "split_by_run_result_delimited",
    ],
    "description": "Split by a result in the flow",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
      "cases",
    ],
    "name": "Split by flow result",
    "type": "split_by_run_result",
  },
  "split_by_scheme": Object {
    "description": "Split by URN type",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Split by URN type",
    "type": "split_by_scheme",
  },
  "split_by_subflow": Object {
    "aliases": Array [
      "split_by_subflow",
    ],
    "component": [Function],
    "description": "Enter another flow",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Enter another flow",
    "type": "enter_flow",
  },
  "split_by_ticket": Object {
    "aliases": Array [
      "split_by_ticket",
    ],
    "component": [Function],
    "description": "Open a ticket with a human agent",
    "filter": "ticketer",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Open Ticket",
    "type": "open_ticket",
  },
  "split_by_webhook": Object {
    "aliases": Array [
      "split_by_webhook",
    ],
    "component": [Function],
    "description": "Call a webhook",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Call webhook",
    "type": "call_webhook",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  "split_by_wenigpt": Object {
    "aliases": Array [
      "split_by_wenigpt",
    ],
    "component": [Function],
    "description": "Call WeniGPT",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "input",
    ],
    "name": "Call WeniGPT",
    "new": true,
    "type": "call_wenigpt",
  },
  "split_by_whatsapp_product": Object {
    "aliases": Array [
      "split_by_whatsapp_product",
    ],
    "component": [Function],
    "description": "Send a WhatsApp product",
    "filter": "whatsapp_catalog",
    "form": [Function],
    "name": "Send WhatsApp Product",
    "new": true,
    "type": "send_msg_catalog",
  },
  "start_session": Object {
    "component": [Function],
    "description": "Start somebody else in a flow",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Start somebody else",
    "type": "start_session",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  "switch": Object {
    "aliases": Array [
      "switch",
    ],
    "description": "Wait for the contact to respond",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "categories",
      "cases",
    ],
    "name": "Wait for response",
    "type": "wait_for_response",
    "visibility": Array [
      "messaging",
      "messaging_offline",
    ],
  },
  "transfer_airtime": Object {
    "aliases": Array [
      "split_by_airtime",
    ],
    "component": [Function],
    "description": "Send the contact airtime",
    "filter": "airtime",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Send Airtime",
    "type": "transfer_airtime",
    "visibility": Array [
      "messaging",
      "messaging_background",
      "voice",
    ],
  },
  "wait_for_audio": Object {
    "description": "Wait for an audio recording",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Wait for Audio",
    "type": "wait_for_audio",
    "visibility": Array [
      "messaging_offline",
      "voice",
    ],
  },
  "wait_for_dial": Object {
    "description": "Wait for forwarded call to someone else",
    "form": [Function],
    "name": "Wait for Forwarded Call",
    "type": "wait_for_dial",
    "visibility": Array [
      "voice",
    ],
  },
  "wait_for_digits": Object {
    "description": "Wait for multiple digits",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
      "cases",
    ],
    "name": "Wait for Digits",
    "type": "wait_for_digits",
    "visibility": Array [
      "voice",
    ],
  },
  "wait_for_image": Object {
    "description": "Wait for an image",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Wait for Image",
    "type": "wait_for_image",
    "visibility": Array [
      "messaging_offline",
    ],
  },
  "wait_for_location": Object {
    "description": "Wait for location GPS coordinates",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Wait for Location",
    "type": "wait_for_location",
    "visibility": Array [
      "messaging_offline",
    ],
  },
  "wait_for_menu": Object {
    "description": "Wait for menu selection",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Wait for Menu Selection",
    "type": "wait_for_menu",
    "visibility": Array [
      "voice",
    ],
  },
  "wait_for_response": Object {
    "aliases": Array [
      "switch",
    ],
    "description": "Wait for the contact to respond",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "categories",
      "cases",
    ],
    "name": "Wait for response",
    "type": "wait_for_response",
    "visibility": Array [
      "messaging",
      "messaging_offline",
    ],
  },
  "wait_for_video": Object {
    "description": "Wait for a video",
    "form": [Function],
    "localization": [Function],
    "localizeableKeys": Array [
      "exits",
    ],
    "name": "Wait for Video",
    "type": "wait_for_video",
    "visibility": Array [
      "messaging_offline",
    ],
  },
}
`;
